Readme.md

```diff
@@ -0,0 +1,181 @@
+
+```bash
+az ad sp create-for-rbac --scopes /subscriptions/mySubscriptionID \
+  --name myServicePrincipalName \
+  --role rolename
+
+az ad sp create-for-rbac --scopes "/subscriptions/mySubscriptionID" --role "Contributor" --name mySpName
+{
+  "appId": "appIdGuid",
+  "displayName": "mySpName",
+  "name": "guid",
+  "password": "password",
+  "tenant": "myTenantId"
+}
+```
+
+You will also need the object id of the SP. You can get that with this Azure CLI call
+
+```bash
+az ad sp show --id <appIdGuid from above> --query id
+```
+
+### Creating SSH key pair
+
+There are a variety of ways to create the SSH pey pair:
+
+- [Generate and store SSH keys in the Azure portal](https://docs.microsoft.com/azure/virtual-machines/ssh-keys-portal)
+- [Create and manage SSH keys locally](https://docs.microsoft.com/azure/virtual-machines/linux/create-ssh-keys-detailed)
+- [Create and manage SSH Keys with the Azure CLI](https://docs.microsoft.com/azure/virtual-machines/ssh-keys-azure-cli)
+
+## Deploying
+
+### Deploying Azure Front Door Standard
+
+Use the following command to deploy the Azure Front Door Standard deployment
+
+```bash
+az deployment sub create --template-file deployRgFrontDoorAndWeb.bicep \
+  --location centralus \
+  -p frontDoorSkuName=Standard_AzureFrontDoor assetPrefix=uniqueprefixname
+```
+
+### Deploying Azure Front Door Premium With Private Endpoints
+
+Use the following command to deploy the Azure Front Door Premium deployment
+
+```bash
+az deployment sub create --template-file deployRgFrontDoorAndWeb.bicep \
+  --location centralus \
+  -p frontDoorSkuName=Premium_AzureFrontDoor assetPrefix=uniqueprefixname
+```
+
+### Deploying the VNet for the Front Door Premium deployment
+
+You will need to update the following parameters in parameters.json:
+
+- jumpboxPublicSshKey - This is the public SSH key you created earlier
+- assetPrefix - This is used as the basis for the names of every asset deployed. For example if the assetPrefix is 'testworkload', the Azure Resource Group name will be 'testworkload-rg'. This assetPrefix MUST match the assetPrefix used in the Front Door Premium deployment
+- storageAccountWebsiteLocations - This is an array of valid Azure locations where Storage Accounts will be created.
+- vmadmin - The name of the admin account to create for the Jumpbox Virtual Machine.
+- principalId - The object id of the Service Principal created above. Do not use the appId.
+
+The following is an example of storageAccountWebsiteLocations:
+
+```bash
+"storageAccountWebsiteLocations": {
+    "value": [
+    "eastus",
+    "westus3"
+    ]
+},
+```
+
+Use the following command to deploy the Virtual Network deployment
+
+```bash
+az deployment group create \
+  --template-file deployVnet.bicep \
+  --resource-group resourceGroupName \
+  --parameters @parameters.json 

```

Comment:
Instead of having someone edit a file, can you instead consider just putting these params inline and populating them with az cli-obtained values instead.  Then they can't mess up (typo, misunderstanding, mismatch of assetPrefix, etc)